### PR TITLE
bulbs-lightbox element

### PR DESCRIPTION
**Pull request by jmelvnsn**
@theonion/front-end 

Lightbox element to replace old Onion implementation to enlarge article images.

https://github.com/theonion/omni/pull/915

### CODE REVIEW

File: elements/bulbs-lightbox/bulbs-lightbox.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          this.classList.add('active');
+        }
+      }
+
+      // Close overlay with Esc key
+      $(document).keyup(function(e) {
+        if (e.keyCode == 27) overlay.classList.remove('active');
+      });
+
+      this.addEventListener('click', toggleOverlay);
```
Comment (collin) [inline elements/bulbs-lightbox/bulbs-lightbox.js]: I think we ought to translate these event listeners to be more consistent with the bulbs-elements house style.

Something like:

```js
document.addEventListener('keyup', this.handleKeyUp.bind(this));
document.addEventListener('click', this.toggleOverlay.bind(this));
```

with `toggleOverlay` and `handleKeyUp` defined as methods on element class.

File: elements/bulbs-lightbox/bulbs-lightbox.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    height: 100vh;
+    justify-content: center;
+    left: 0;
+    opacity: 0;
+    position: fixed;
+    right: 0;
+    top: 0;
+    transform: scale(0.9);
+    transition: transform 0.2s, opacity 0.2s, visibility 0s 0.2s;
+    visibility: hidden;
```
Comment (brycedorn) [inline elements/bulbs-lightbox/bulbs-lightbox.scss]: looks like this [won't load the img asynchronously](http://stackoverflow.com/questions/578283/will-an-image-load-if-it-has-its-css-set-so-that-visibility-is-hidden), is that intended?

I made a library kinda like this a little while ago - will help with page load if just the preview is loaded (and the larger version onclick)

File: elements/bulbs-lightbox/bulbs-lightbox-examples.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import './bulbs-lightbox.scss';
+import './bulbs-lightbox-test.scss';
+
+let examples = {
+  element: 'bulbs-lightbox',
+  examples: {
+    'Bulbs LightBox': {
+      render: () => {
+        return `
+          <bulbs-lightbox>
```
Comment (spra85) [inline elements/bulbs-lightbox/bulbs-lightbox-examples.js]: @jmelvnsn opted to make this a semantic component as opposed to `is="bulbs-lightbox"`

Comment (jmelvnsn) [inline elements/bulbs-lightbox/bulbs-lightbox-examples.js]: yeah. little easier to read/comprehend in the document source

**Pull request merged**